crates/iota/src/client_commands.rs

```diff
@@ -2916,6 +2950,53 @@ pub(crate) async fn dry_run_or_execute_or_serialize(
     }
 }
 
+async fn execute_dev_inspect(
+    context: &mut WalletContext,
+    signer: IotaAddress,
+    tx_kind: TransactionKind,
+    gas_budget: Option<u64>,
+    gas_price: u64,
+    gas_payment: Option<Vec<ObjectID>>,
+    gas_sponsor: Option<IotaAddress>,
+    skip_checks: Option<bool>,
+) -> Result<IotaClientCommandResult, anyhow::Error> {
+    let client = context.get_client().await?;
+    let gas_budget = gas_budget.map(iota_serde::BigInt::from);
+    let mut gas_objs = vec![];
```

Comment:
Too late but with your change, you can move that in the else to avoid a potential useless allocation. Maybe do it in another unrelated PR.